src/Stripe.Tests.XUnit/sources/listing_sources_on_customer.cs

```diff
@@ -0,0 +1,101 @@
+using FluentAssertions;
+using Xunit;
+
+namespace Stripe.Tests.Xunit
+{
+    public class listing_sources_on_customer
+    {
+        public StripeList<StripeSource> SourceListAll { get; }
+        public StripeList<StripeSource> SourceListCard { get; }
+        public StripeList<StripeSource> SourceListBitcoin { get; }
+
+        public listing_sources_on_customer()
+        {
+            var sourceService = new StripeSourceService(Cache.ApiKey);
+            var customerService = new StripeCustomerService(Cache.ApiKey);
+
+            // Create customer
+            var CustomerCreateOptions = new StripeCustomerCreateOptions
+            {
+                Email = "source-list@example.com",
+            };
+            var Customer = customerService.Create(CustomerCreateOptions);
+
+            // Create card source and attach it to customer
+            var SourceCardCreateOptions = new StripeSourceCreateOptions
+            {
+                Type = StripeSourceType.Card,
+                Token = "tok_visa"
+            };
+            var SourceCard = sourceService.Create(SourceCardCreateOptions);
+
+            var SourceAttachOptions = new StripeSourceAttachOptions
+            {
+                Source = SourceCard.Id
+            };
+            SourceCard = sourceService.Attach(Customer.Id, SourceAttachOptions);
+
+            // Create bitcoin source and attach it to customer
+            var SourceBitcoinCreateOptions = new StripeSourceCreateOptions
+            {
+                Type = StripeSourceType.Bitcoin,
+                Amount = 1000,
+                Currency = "usd",
+                Owner = new StripeSourceOwner
+                {
+                    Email = "jenny.rosen+fill_now@example.com",
+                },
+            };
+            var SourceBitcoin = sourceService.Create(SourceBitcoinCreateOptions);
+
+            SourceAttachOptions.Source = SourceBitcoin.Id;
```

Comment:
Minor but I like having separate options instances. Jayme used to compare the type on the object to the info on the options object itself but here since you override you can't do this.

It's definite more verbose my way but I found it cleaner when I started to work on the library. Feel free to ignore

Reply:
Oops, not sure why I missed this comment earlier.

Yeah, if you need the options object in your test it's better to instantiate it separately. In this case the options object are only instantiated within the test constructor though, and only the resulting list objects are persisted and used in test assertions.